src/sentry/static/sentry/app/views/settings/components/dataPrivacyRulesPanel/dataPrivacyRulesPanelForm/dataPrivacyRulesPanelFormEventId.tsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import isEqual from 'lodash/isEqual';
+
+import TextField from 'app/components/forms/textField';
+import {t} from 'app/locale';
+import space from 'app/styles/space';
+import Alert from 'app/components/alert';
+import ControlState from 'app/views/settings/components/forms/field/controlState';
+import {addErrorMessage, addSuccessMessage} from 'app/actionCreators/indicator';
+
+import DataPrivacyRulesPanelFormField from './dataPrivacyRulesPanelFormField';
+import {EVENT_ID_FIELD_STATUS} from '../utils';
+
+type EventId = {
+  value: string;
+  status?: EVENT_ID_FIELD_STATUS;
+};
+
+type Props = {
+  onUpdateEventId: (eventId: string) => void;
+  eventId: EventId;
+  disabled?: boolean;
+};
+
+const loadEventIdStatus = (status?: EVENT_ID_FIELD_STATUS) => {
+  switch (status) {
+    case EVENT_ID_FIELD_STATUS.INVALID:
+      addErrorMessage(t("That's not a valid event ID"));
+      break;
+    case EVENT_ID_FIELD_STATUS.ERROR:
+      addErrorMessage(t('Something went wrong while fetching the suggestions'));
+      break;
+    case EVENT_ID_FIELD_STATUS.NOT_FOUND:
+      addErrorMessage(t('Event ID not found in projects you have access to'));
+      break;
+    case EVENT_ID_FIELD_STATUS.LOADED:
+      addSuccessMessage(
+        t('Successfully loaded event for autocompletion')
+      );
+      break;
+    default:
+  }
+};
+
+type State = {
+  value: string;
+  status?: EVENT_ID_FIELD_STATUS;
+};
+
+class DataPrivacyRulesPanelFormEventId extends React.Component<Props, State> {
+  state = {
+    value: this.props.eventId.value,
+    status: this.props.eventId?.status,
+  };
+
+  componentDidUpdate(prevProps: Props) {
+    if (!isEqual(prevProps.eventId, this.props.eventId)) {
+      this.loadState();
+    }
+  }
+
+  loadState = () => {
+    this.setState(
+      {
+        ...this.props.eventId,
+      },
+      () => {
+        this.loadStatus();
+      }
+    );
+  };
+
+  loadStatus = () => {
+    loadEventIdStatus(this.state.status);
+  };
+
+  handleChange = (value: string) => {
+    const eventId = value.replace(/-/g, '').trim();
+
+    if (eventId !== this.state.value) {
+      this.setState({
+        value: eventId,
+        status: undefined,
+      });
+    }
+  };
+
+  isEventIdValid = (): boolean => {
+    const {value} = this.state;
+
+    if (value && value.length !== 32) {
+      this.setState({status: EVENT_ID_FIELD_STATUS.INVALID}, () => {
+        this.loadStatus();
+      });
+      return false;
+    }
+
+    return true;
+  };
+
+  handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+    event.preventDefault();
+
+    if (this.isEventIdValid()) {
+      this.props.onUpdateEventId(this.state.value);
+    }
+  };
+
+  handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    event.persist();
```

Comment:
Why persist the event?

Reply:
I am not sure if it was me who added it here..anyway I just removed it...good catch...thanks :)